File: csharp/playwright-basic/Scraper.cs
using Microsoft.Playwright;

class Scraper
{

    private IPlaywright _pw;
    private string _auth;

    public Scraper(IPlaywright pw, string auth)
    {
        _pw = pw;
        _auth = auth;
    }

    private async Task<IBrowser> Connect()
    {
        if (_auth == "USER:PASS")
        {
            throw new Exception("Provide Scraping Browsers credentials in AUTH"
                    + " environment variable or update the script.");
        }
        var endpointURL = $"wss://{_auth}@brd.superproxy.io:9222";
        return await _pw.Chromium.ConnectOverCDPAsync(endpointURL);
    }

    public async Task Scrape(string url)
    {
        Log("Connecting to Browser...");
        var browser = await Connect();
        try {
            Log($"Connected! Navigating to {url}...");
            var page = await browser.NewPageAsync();
            await page.GotoAsync(url, new (){ Timeout = 2 * 60 * 1000 });
            Log("Navigated! Scraping page content...");
            var data = await page.ContentAsync();
            Log($"Scraped! Data: {data}");
        } finally {
            await browser.CloseAsync();
        }
    }

    private static string Env(string name, string defaultValue)
    {
        return Environment.GetEnvironmentVariable(name) ?? defaultValue;
    }

    private static void Log(string message)
    {
        Console.WriteLine(message);
    }

    public static async Task Main()
    {
        var auth = Env("AUTH", "USER:PASS");
        var url = Env("TARGET_URL", "https://example.com");
        var pw = await Playwright.CreateAsync();
        var scraper = new Scraper(pw, auth);
        await scraper.Scrape(url);
    }

}

File: java/playwright-basic/src/main/java/Scraper.java
import com.microsoft.playwright.*;
import java.util.HashMap;

public class Scraper {

    private Playwright _pw;
    private String _auth;

    public Scraper(Playwright pw, String auth) {
        _pw = pw;
        _auth = auth;
    }

    private Browser connect() throws Exception {
        if (_auth == "USER:PASS") {
            throw new Exception("Provide Scraping Browsers credentials in AUTH"
                    + " environment variable or update the script.");
        }
        var endpointURL = "wss://" + _auth + "@brd.superproxy.io:9222";
        return _pw.chromium().connectOverCDP(endpointURL);
    }

    public void scrape(String url) throws Exception {
        log("Connecting to Browser...");
        var browser = connect();
        try {
            log("Connected! Navigating to %s...", url);
            var page = browser.newPage();
            page.navigate(url);
            log("Navigated! Scraping page content...");
            var data = page.content();
            log("Scraped! Data: %s", data);
        } finally {
            browser.close();
        }
    }

    private static void log(String message, Object... params) {
        System.out.println(String.format(message, params));
    }

    public static void main(String[] args) throws Exception {
        var env = System.getenv();
        var auth = env.getOrDefault("AUTH", "USER:PASS");
        var targetUrl = env.getOrDefault("TARGET_URL", "https://example.com");
        var options = new Playwright.CreateOptions()
            .setEnv(new HashMap<>(env){{
                put("PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD", "1");
            }});
        try (var pw = Playwright.create(options)) {
            var scraper = new Scraper(pw, auth);
            scraper.scrape(targetUrl);
        }
    }

}
